### PR TITLE
MISUV-2383 - Migrate Next Updates pages

### PR DESCRIPTION
**Pull request by Wright93**
- Refactored ReportDeadlines to NextUpdates

### CODE REVIEW

File: app/audit/models/NextUpdatesResponseAuditModel.scala [hunk 1]
```diff
@@ -17,32 +17,32 @@
...
 
-  private val reportDeadlinesJson: Seq[JsObject] = reportDeadlines.map(reportDeadineJson)
+  private val nextUpdatesJson: Seq[JsObject] = nextUpdates.map(nextUpdateJson)
 
   override val detail: JsValue = Json.obj(
     "mtditid" -> mtdItUser.mtditid,
     "nationalInsuranceNumber" -> mtdItUser.nino,
     "incomeSourceId" -> incomeSourceId,
-    "reportDeadlines" -> reportDeadlinesJson
+    "nextUpdates" -> nextUpdatesJson
```
Comment (MihailsC-hmrc) [inline app/audit/models/NextUpdatesResponseAuditModel.scala:45]: Suspicious change: has the renaming of TxM audit json field `"reportDeadlines"` been approved? This change will be visible in Splunk logs. Please double check this change with the Tech Lead.

File: app/connectors/IncomeTaxViewChangeConnector.scala [hunk 2]
```diff
@@ -63,7 +63,7 @@ trait IncomeTaxViewChangeConnector extends RawResponseReads with FeatureSwitchin
     s"${appConfig.itvcProtectedService}/income-tax-view-change/nino-lookup/$mtdRef"
   }
 
-  def getReportDeadlinesUrl(nino: String): String = {
+  def getNextUpdatesUrl(nino: String): String = {
```
Comment (MihailsC-hmrc) [inline app/connectors/IncomeTaxViewChangeConnector.scala:66]: Just a note: with this change the url and the name of the function go out of sync. Url says `report-deadlines`, the function says `NextUpdates`

File: app/controllers/HomeController.scala [hunk 3]
```diff
@@ -38,11 +38,11 @@ import scala.concurrent.{ExecutionContext, Future}
 
 @Singleton
 class HomeController @Inject()(val homeView: views.html.Home,
-                                val checkSessionTimeout: SessionTimeoutPredicate,
+                               val checkSessionTimeout: SessionTimeoutPredicate,
                                val authenticate: AuthenticationPredicate,
                                val retrieveNino: NinoPredicate,
                                val retrieveIncomeSources: IncomeSourceDetailsPredicate,
-                               val reportDeadlinesService: ReportDeadlinesService,
+                               val NextUpdatesService: NextUpdatesService,
```
Comment (MihailsC-hmrc) [inline app/controllers/HomeController.scala:45]:
```suggestion
                               val nextUpdatesService: NextUpdatesService,
```

File: app/services/NextUpdatesService.scala [hunk 4]
```diff
@@ -16,29 +16,30 @@
...
 import connectors._
+import javax.inject.{Inject, Singleton}
 import models.incomeSourceDetails.IncomeSourceDetailsModel
-import models.reportDeadlines._
+import models.nextUpdates._
 import play.api.Logger
 import uk.gov.hmrc.http.{HeaderCarrier, InternalServerException}
 
-import java.time.LocalDate
-import javax.inject.{Inject, Singleton}
```
Comment (MihailsC-hmrc) [inline app/services/NextUpdatesService.scala:27]: Just a note: probably imports were optimised in some older version of Intellij IDEA or its Scala plugin. The newer versions put standard java/scala imports from top to here, the bottom of imports section :)

File: app/views/agent/NextUpdates.scala.html [hunk 5]
```diff
@@ -19,7 +19,7 @@
 
 @this(mainTemplate: MainTemplate, nextUpdatesHelper: NextUpdatesHelper)
```
Comment (MihailsC-hmrc) [inline app/views/agent/NextUpdates.scala.html:20]: Question: Does the agent page look okay still? It hasn't been migrated yet to new layout but already uses the "migrated" `NextUpdatesHelper` with new classes there. Is everything okay on the agent page?

File: app/views/NextUpdates.scala.html [hunk 6]
```diff
@@ -44,32 +49,29 @@ <h1 id="page-heading" class="heading-xlarge"> @messages("nextUpdates.heading")</
...
 
 		<div class="govuk-details__text" id="details-content-0" data-module="govuk-details" aria-hidden="false">
-			<h2 id="quarterlyReturns-heading" class="heading-medium">@messages("obligations.quarterlyUpdates")</h2>
-			<p><span id="quarterly-dropdown-line1">@messages("nextUpdates.dropdown.quarterlyReturn.text")</span> </p>
-			<p><span id="quarterly-dropdown-line2">@Html(messages("nextUpdates.dropdown.quarterlyReturn.text.lin2",
-			))
-			</span>
-			</p>
+			<h2 id="quarterlyReturns-heading" class="govuk-heading-m">@messages("obligations.quarterlyUpdates")</h2>
+			@p(id = Some("quarterly-dropdown-line1"), content = Html(messages("nextUpdates.dropdown.quarterlyReturn.text")))
```
Comment (MihailsC-hmrc) [inline app/views/NextUpdates.scala.html:53]: Note: To my mind having the component like `p` which requires to write `Html(...)` in the view is evil. I would say we better make the `p.apply(...)` curried and the content html is better to be passed there solely in a last argument list, like:
```
@p(id = Some("quarterly-dropdown-line1")) { 
    @messages("nextUpdates.dropdown.quarterlyReturn.text")
}
```

File: app/views/NextUpdates.scala.html [hunk 7]
```diff
@@ -44,32 +49,29 @@ <h1 id="page-heading" class="heading-xlarge"> @messages("nextUpdates.heading")</
...
+			@p(id = Some("quarterly-dropdown-line2"), content = Html(messages("nextUpdates.dropdown.quarterlyReturn.text.lin2")))
```
Comment (MihailsC-hmrc) [inline app/views/NextUpdates.scala.html:54]: I don't see `<span` in the content here. Is it not needed?

File: it/helpers/servicemocks/IncomeTaxViewChangeStub.scala [hunk 8]
```diff
@@ -95,21 +95,21 @@ object IncomeTaxViewChangeStub { // scalastyle:off number.of.methods
...
 
-  def stubGetReportDeadlinesNotFound(nino: String): Unit =
-    WiremockHelper.stubGet(reportDeadlinesUrl(nino), Status.NO_CONTENT, "")
+  def stubgetNextUpdatesNotFound(nino: String): Unit =
+    WiremockHelper.stubGet(nextUpdatesUrl(nino), Status.NO_CONTENT, "")
 
-  def verifyGetReportDeadlines(nino: String): Unit =
-    WiremockHelper.verifyGet(reportDeadlinesUrl(nino))
+  def verifygetNextUpdates(nino: String): Unit =
+    WiremockHelper.verifyGet(nextUpdatesUrl(nino))
```
Comment (MihailsC-hmrc) [inline it/helpers/servicemocks/IncomeTaxViewChangeStub.scala:112]: Please rename functions with camelCase names i.e. `stubGetNextUpdates`, `stubGetNextUpdatesError` etc.

File: test/controllers/NextUpdatesControllerSpec.scala [hunk 9]
```diff
@@ -16,76 +16,77 @@
...
 import scala.concurrent.Future
 
-class ReportDeadlinesControllerSpec extends MockAuthenticationPredicate with MockIncomeSourceDetailsPredicate
-                                            with MockReportDeadlinesService with FeatureSwitching{
+class NextUpdatesControllerSpec extends MockAuthenticationPredicate with MockIncomeSourceDetailsPredicate
+                                            with MockNextUpdatesService with FeatureSwitching{
 
-  object TestReportDeadlinesController extends ReportDeadlinesController(
-    app.injector.instanceOf[NoReportDeadlines],
+  object TestNextUpdatesController$ extends NextUpdatesController(
```
Comment (MihailsC-hmrc) [inline test/controllers/NextUpdatesControllerSpec.scala:45]: Please rename the object to get rid of the '$' sign at the end

**Review by MihailsC-hmrc — COMMENTED** [review]
Can't approve, need more clarifications and probably some fixes there.

File: it/helpers/servicemocks/IncomeTaxViewChangeStub.scala (hunk 8 above)
Comment (stephenkward) [reply it/helpers/servicemocks/IncomeTaxViewChangeStub.scala]: Fixed

File: test/controllers/NextUpdatesControllerSpec.scala (hunk 9 above)
Comment (stephenkward) [reply test/controllers/NextUpdatesControllerSpec.scala]: Fixed

File: app/controllers/HomeController.scala (hunk 3 above)
Comment (stephenkward) [reply app/controllers/HomeController.scala]: Fixed

File: app/audit/models/NextUpdatesResponseAuditModel.scala (hunk 1 above)
Comment (Wright93) [reply app/audit/models/NextUpdatesResponseAuditModel.scala]: Changed back to "reportDeadlines"

File: app/connectors/IncomeTaxViewChangeConnector.scala (hunk 2 above)
Comment (Wright93) [reply app/connectors/IncomeTaxViewChangeConnector.scala]: Changed back to "getReportDeadlinesUrl"

File: app/views/agent/NextUpdates.scala.html (hunk 5 above)
Comment (Wright93) [reply app/views/agent/NextUpdates.scala.html:20]: Checked agent NextUpdates, added in a extra "visuallyhidden" class to the "Open all" to hide the "sections" text that was appearing. Looks visually fine

File: app/views/NextUpdates.scala.html (hunk 7 above)
Comment (Wright93) [reply app/views/NextUpdates.scala.html:54]: No <span> in the prototype so removed from the frontend, looks visually fine, and no accessibility errors from Wave

File: app/views/NextUpdates.scala.html (hunk 6 above)
Comment (MihailsC-hmrc) [reply app/views/NextUpdates.scala.html:53]: Not needed to be solved right now, just a side note to think about, no worries.

**Review by MihailsC-hmrc — APPROVED** [review]

**Pull request merged**